imagelab/ImageLab.java

```diff
@@ -288,13 +288,15 @@ public void actionPerformed(final ActionEvent e) {
                 fd = new FileDialog(frame, "Pick an image", FileDialog.LOAD);
                 fd.setVisible(true);
                 String theFile = fd.getFile();
-                String theDir = fd.getDirectory();
-                //System.out.println("The file's name is " + theDir + theFile);
-                improvider = new ImgProvider(theDir + theFile);
-                improvider.setLab(theLab);
-                improvider.showImage(theDir + theFile);
-                images.add(improvider);
-                impro = improvider; //current image provider is set
+                if (theFile != null) {
+                    String theDir = fd.getDirectory();
+                    //System.out.println("The file's name is " + theDir + theFile);
```

Comment:
Let's clean up this code by deleting this commented-out "debugging" line at the same time as making the fix.